sdk/storage/azure-storage-data-movement/src/main/java/com/azure/storage/data/movement/PathScanner.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.data.movement;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.nio.file.*;
+import java.util.stream.BaseStream;
+
+public class PathScanner {
+    private final ClientLogger logger = new ClientLogger(PathScanner.class);
+    private final String basePath;
+    private final boolean isDir;
+
+    public PathScanner(String path) {
+        // Resolve the given path to an absolute path in case it isn't one already
+        Path pathObj = Paths.get(path);
+        basePath = pathObj.toAbsolutePath().toString();
+
+        // Check if the path exists and whether or not it's a directory; throw
+        // an error if there's nothing present or readable at the given path
+        if (Files.exists(pathObj)) {
+            isDir = Files.isDirectory(pathObj);
+        } else {
+            throw logger.logExceptionAsError(new IllegalArgumentException(String.format("No accessible item exists at the path '%s'.", basePath)));
+        }
+    }
+
+    public Flux<String> scan(boolean skipSubdirectories) {
+        // Iterate through the path if it's a directory
+        if (isDir) {
```

Comment:
I wonder if we even need the if statement for isDir at this level if we are checking if its a directory on line 42

Reply:
good point, I'll check this with more unit tests and remove if possible

Reply:
Yup, wasn't needed. In fact I was able to take out all related logic, so there is no isDir anymore.